examples/oft-solana/tasks/solana/setUpdateAuthority.ts

```diff
@@ -0,0 +1,99 @@
+
+import { types as devtoolsTypes } from '@layerzerolabs/devtools-evm-hardhat'
+import { promptToContinue } from '@layerzerolabs/io-devtools'
+import { EndpointId } from '@layerzerolabs/lz-definitions'
+
+import { deriveConnection, getExplorerTxLink } from '.'
+
+interface Args {
+    mint: string
+    newUpdateAuthority?: string
+    renounceUpdateAuthority?: boolean
+    eid: EndpointId
+}
+
+// sets the update authority via Metaplex
+task('lz:oft:solana:set-update-authority', 'Updates the metaplex update authority of the SPL Token')
+    .addParam('eid', 'Solana mainnet (30168) or testnet (40168)', undefined, devtoolsTypes.eid)
+    .addParam('mint', 'The Token mint public key', undefined, devtoolsTypes.string)
+    .addOptionalParam('newUpdateAuthority', 'The new update authority', undefined, devtoolsTypes.string)
+    .addOptionalParam('renounceUpdateAuthority', 'Renounce update authority', false, devtoolsTypes.boolean)
+    .setAction(
+        async ({ eid, mint: mintStr, newUpdateAuthority: newUpdateAuthorityStr, renounceUpdateAuthority }: Args) => {
+            // if not renouncing, must provide new update authority
+            if (!renounceUpdateAuthority && !newUpdateAuthorityStr) {
+                throw new Error(
+                    'Either specify the new update authority via --new-update-authority <NEW_UPDATE_AUTHORITY> or renounce via --renounce-update-authority true'
+                )
+            }
+
+            // if renouncing, must not provide new update authority
+            if (renounceUpdateAuthority && newUpdateAuthorityStr) {
+                throw new Error('Cannot provide new update authority if renouncing')
+            }
+
+            /*
+             * On why the update authority is set to SystemProgram.programId ("11111111111111111111111111111111") when renouncing:
+             * The Metaplex Token Metadata program defines the update_authority strictly as a Pubkey:
+             * https://github.com/metaplex-foundation/mpl-token-metadata/blob/23aee718e723578ee5df411f045184e0ac9a9e63/programs/token-metadata/program/src/state/metadata.rs#L73
+             * Hence, the value must always be a Pubkey
+             * To renounce the update authority, we can to set its value to SystemProgram ID ("11111111111111111111111111111111")
+             * This is done on top of setting `isMutable` to false
+             */
+
+            const updateAuthority = renounceUpdateAuthority
+                ? publicKey(SystemProgram.programId)
+                : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                  publicKey(newUpdateAuthorityStr!) // we already checked that this is defined
+
+            const { umi, umiWalletSigner } = await deriveConnection(eid)
+
+            const mint = publicKey(mintStr)
+            const initialMetadata = await fetchMetadataFromSeeds(umi, { mint })
+
+            if (initialMetadata.updateAuthority == SystemProgram.programId.toString()) {
```

Comment:
nit: use `===` instead of `==` for strict equality checking.

also you're doing a strict check in the next if statement right after this, so better to be consistent

Reply:
thanks!